File: internal/content/testdata/bench/go/adjacent-pair-compression.go
func compressed_length(s string) int {
	n := len(s)
	count := 0
	i := 0
	for i < n {
		if i+1 < n && s[i] == s[i+1] {
			count++
			i += 2
		} else {
			count++
			i++
		}
	}
	return count
}

File: internal/content/testdata/bench/rust/adjacent-pair-compression.rs
fn compressed_length(s: String) -> i64 {
    let chars: Vec<char> = s.chars().collect();
    let n = chars.len();
    let mut count = 0i64;
    let mut i = 0usize;
    while i < n {
        if i + 1 < n && chars[i] == chars[i + 1] {
            count += 1;
            i += 2;
        } else {
            count += 1;
            i += 1;
        }
    }
    count
}
